app/services/GetIncomeSourcesService.scala

```diff
@@ -16,24 +16,30 @@
 
 package services
 
+import connectors.{IncomeTaxDividendsConnector, IncomeTaxInterestConnector}
 import javax.inject.Inject
-import uk.gov.hmrc.http.HeaderCarrier
-import connectors.IncomeTaxDividendsConnector
-import models.{DividendsResponseModel, ErrorResponse, IncomeSourcesResponseModel, SubmittedDividendsModel}
+import models._
 import services.util.FutureEitherOps
+import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class GetIncomeSourcesService @Inject()(dividendsConnector: IncomeTaxDividendsConnector, implicit val ec: ExecutionContext) {
+class GetIncomeSourcesService @Inject()(dividendsConnector: IncomeTaxDividendsConnector,
+                                        interestConnector: IncomeTaxInterestConnector,
+                                        implicit val ec: ExecutionContext) {
 
   type IncomeSourceResponse = Either[ErrorResponse, IncomeSourcesResponseModel]
 
   def getAllIncomeSources(nino: String, taxYear: Int, mtditid: String)(implicit hc: HeaderCarrier): Future[IncomeSourceResponse] =  {
 
     val incomeSources = for {
       dividends <- FutureEitherOps[ErrorResponse, Option[SubmittedDividendsModel]](dividendsConnector.getSubmittedDividends(nino, taxYear, mtditid))
+      interest <- FutureEitherOps[ErrorResponse, Option[SubmittedInterestModel]](interestConnector.getSubmittedInterest(nino, taxYear, mtditid))
+
     } yield {
-      IncomeSourcesResponseModel(dividends.map(res => DividendsResponseModel(res.ukDividends, res.otherUkDividends)))
+
+      IncomeSourcesResponseModel(dividends.map(res => DividendsResponseModel(res.ukDividends, res.otherUkDividends)),
```

Comment:
Converting them from a model into an identical model feels redundant to me. Not going to ask you to change it, but definitely something we can keep an eye on.